node/src/components/consensus/era_supervisor.rs

```diff
@@ -131,17 +136,22 @@ pub struct Era<I> {
     /// Pending candidate blocks, waiting for validation. The boolean is `true` if the proto block
     /// has been validated; the vector contains the list of accused validators missing evidence.
     candidates: Vec<PendingCandidate>,
+    /// Validators banned in this and the next BONDED_ERAS eras, because they were slashed in the
+    /// previous switch block.
+    banned: Vec<PublicKey>,
```

Comment:
When I read this comment I'm wondering – how do you track in which era those validators will stop being "carried over", how do you collect `banned` validators from subsequent eras… Maybe we can delete the part "and the next BONDED_ERAS eras" and comment on it on the call sides – where validators for the next era are being collected from past ones.

Reply:
We initialize each new `Highway` instance's list of banned validators by collecting the `banned` fields from all still-bonded eras. The `banned` field itself just contains whoever was accused in the previous switch block (which currently is everyone who equivocated within `BONDED_ERAS` _of that switch block_, effectively doubling the banning period; but that will be addressed in [NDRS-495](https://casperlabs.atlassian.net/browse/NDRS-495)).

Would it be better to rename it to `newly_banned` or `newly_slashed`?

Reply:
Also, I realized I forgot to ban the most recently slashed validators. I only banned the ones from the switch block before that, and the one before that, etc.
Will fix that, too…

Reply:
Here goes my review …